### DISCUSSION TITLE
how to disable "clean_dirs" in driver.quit() function

### DISCUSSION THREAD

**Discussion by henzycuong1** [kaliiiiiiiiii/Selenium-Driverless#57] [Q&A] [answered]
I tried example code and got an inconvenience is the "quit()" function will delete all dir even when i used "user-data-dir"
```python
async def quit(self, timeout: float = 30) -> None:
        """Quits the target and closes every associated window.

        :Usage:
            ::

                target.quit()
        """

        def clean_dirs(dirs: typing.List[str]):
            for _dir in dirs:
                while os.path.isdir(_dir):
                    shutil.rmtree(_dir, ignore_errors=True)

        start = time.monotonic()
        # noinspection PyUnresolvedReferences,PyBroadException
        try:
            await self.base_target.execute_cdp_cmd("Browser.close")
        except websockets.exceptions.ConnectionClosedError:
            pass
        except Exception:
            import sys
            print('Ignoring exception at self.base_target.execute_cdp_cmd("Browser.close")', file=sys.stderr)
            traceback.print_exc()

        if not self._is_remote:
            # noinspection PyBroadException,PyUnusedLocal
            try:
                # wait for process to be killed
                while True:
                    try:
                        os.kill(self.browser_pid, 15)
                    except OSError:
                        break
                    await asyncio.sleep(0.1)
                    check_timeout(start, timeout)
                loop = asyncio.get_running_loop()
                await asyncio.wait_for(
                    # wait for
                    loop.run_in_executor(None,
                                         lambda: clean_dirs([self._temp_dir, self._options.user_data_dir])),
                    timeout=timeout - (time.monotonic() - start))
            except Exception as e:
                traceback.print_exc()
            finally:
                pass
```

can you create a feature except user_data_dir ? Thank you so much

**Comment by ttlns** [comment]
@henzycuong1 That currently isn't supported. I might consider implementing it tho:)
For now, just copy the directory to another path bofore quiting👀

**Comment by ttlns** [comment, marked as answer]
@henzycuong1 
This should be resolved with [`selenium-driverless==1.6`](https://github.com/kaliiiiiiiiii/Selenium-Driverless/discussions/75)

**Reply by wealthr** [reply]
i'm getting error driver.quit(clean_dirs=False)
TypeError: Chrome.quit() got an unexpected keyword argument 'clean_dirs'.


and the browser doesn't quit

**Reply by ttlns** [reply]
What version? What script?

**Reply by wealthr** [reply]
SELENIUM-DRIVERLESS==1.6

CODE `from selenium_profiles.webdriver import Chrome
from selenium_profiles.profiles import profiles
from selenium_driverless.webdriver import ChromeOptions
from selenium_driverless.types.by import By

profile = profiles.Windows()  # or .Android
options = ChromeOptions()
# options.add_argument("--headless=new")
driver = Chrome(profile, options=options, driverless_options=True)

# get url
driver.get('https://nowsecure.nl#relax')  # test fingerprint

driver.quit(clean_dirs=False) # Execute on the End!`


ERROR  
C:\Users\alima\PycharmProjects\pythonProject\venv\Scripts\python.exe C:\Users\alima\PycharmProjects\pythonProject\TEST.py 
Traceback (most recent call last):
  File "C:\Users\alima\PycharmProjects\pythonProject\TEST.py", line 14, in <module>
    driver.quit(clean_dirs=False) # Execute on the End!
TypeError: Chrome.quit() got an unexpected keyword argument 'clean_dirs'

Process finished with exit code 1

**Reply by ttlns** [reply]
Uh yeah but that's selenium-profiles, which doesn"t support that argument:)

**Reply by wealthr** [reply]
but selenium-profiles also clears the browser dir

**Reply by wealthr** [reply]
any solution for this?

**Reply by ttlns** [reply]
copy the data over before quitiing the driver.
